agent/agent/build.gradle.kts

```diff
@@ -162,8 +162,36 @@ tasks {
     delete(rootProject.file("licenses"))
   }
 
+  val removeLicenseDate by registering {
+    // removing the license report date makes it idempotent
+    doLast {
+      val filePath = rootDir.toPath().resolve("licenses").resolve("licenses.md")
+      if (Files.exists(filePath)) {
+        val datePattern = Pattern.compile("^_[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2} .*_$")
+        val lines = Files.readAllLines(filePath)
+        // 4th line contains the timestamp of when the license report was generated, replace it with
+        // an empty line
+        if (lines.size > 3 && datePattern.matcher(lines[3]).matches()) {
+          lines[3] = ""
+          Files.write(filePath, lines)
+        }
+      }
+    }
+  }
+
+  val generateLicenseReportEnabled =
+    gradle.startParameter.taskNames.any { it.equals("generateLicenseReport") }
   named("generateLicenseReport").configure {
     dependsOn(cleanLicenses)
+    finalizedBy(":spotlessApply")
+    finalizedBy(removeLicenseDate)
+    // disable licence report generation unless this task is explicitly run
+    // the files produced by this task are used by other tasks without declaring them as dependency
+    // which gradle considers an error
+    enabled = enabled && generateLicenseReportEnabled
+  }
+  if (generateLicenseReportEnabled) {
+    project.parent?.tasks?.getByName("spotlessMisc")?.dependsOn(named("generateLicenseReport"))
   }
```

Comment:
copied of https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ee8c94889c01e0e5efae27155b36ee7b1c453fbc/javaagent/build.gradle.kts#L282-L312